src/lib/core/CHIPTLV.h

```diff
@@ -95,77 +86,701 @@ enum
+     *                                      unsigned), or the reader is not positioned on an element.
+     *
+     */
     CHIP_ERROR Get(uint64_t & v);
-    CHIP_ERROR Get(float & v);
+
+    /**
+     * Get the value of the current element as a double-precision floating point number.
+     *
+     * @param[out]  v                       Receives the value associated with current TLV element.
+     *
+     * @retval #CHIP_NO_ERROR              If the method succeeded.
+     * @retval #CHIP_ERROR_WRONG_TLV_TYPE  If the current element is not a TLV floating point type, or
+     *                                      the reader is not positioned on an element.
+     *
+     */
     CHIP_ERROR Get(double & v);
+
+    /**
+     * Get the value of the current element as a single-precision floating point number.
+     *
+     * @param[out]  v                       Receives the value associated with current TLV element.
+     *
+     * @retval #CHIP_NO_ERROR              If the method succeeded.
+     * @retval #CHIP_ERROR_WRONG_TLV_TYPE  If the current element is not a TLV floating point type, or
+     *                                      the reader is not positioned on an element.
+     *
+     */
+    CHIP_ERROR Get(float & v);
+
+    /**
+     * Get the value of the current byte or UTF8 string element.
+     *
+     * To determine the required input buffer size, call the GetLength() method before calling GetBytes().
+     *
+     * @note The data output by this method is NOT null-terminated.
+     *
+     * @param[in]  buf                      A pointer to a buffer to receive the string data.
+     * @param[in]  bufSize                  The size in bytes of the buffer pointed to by @p buf.
+     *
+     * @retval #CHIP_NO_ERROR              If the method succeeded.
+     * @retval #CHIP_ERROR_WRONG_TLV_TYPE  If the current element is not a TLV byte or UTF8 string, or
+     *                                      the reader is not positioned on an element.
+     * @retval #CHIP_ERROR_BUFFER_TOO_SMALL
+     *                                      If the supplied buffer is too small to hold the data associated
+     *                                      with the current element.
+     * @retval #CHIP_ERROR_TLV_UNDERRUN    If the underlying TLV encoding ended prematurely.
+     * @retval other                        Other CHIP or platform error codes returned by the configured
+     *                                      TLVBackingStore.
+     *
+     */
     CHIP_ERROR GetBytes(uint8_t * buf, uint32_t bufSize);
+
+    /**
+     * Allocates and returns a buffer containing the value of the current byte or UTF8 string.
+     *
+     * This method creates a buffer for and returns a copy of the data associated with the byte
+     * or UTF-8 string element at the current position. Memory for the buffer is obtained with
+     * Platform::MemoryAlloc() and should be freed with Platform::MemoryFree() by the caller when
```

Comment:
I wonder if we should fully qualify names such as `chip::Platform::MemoryAlloc` here? Unsure of the conventions on this.

Reply:
This set of comments was merely moved from the .cpp files so I didn't consider it at all. Personally, I agree with leaving off `chip::` inside CHIP.